File: src/main/java/me/zeroeightsix/kami/command/commands/PrefixCommand.java
package me.zeroeightsix.kami.command.commands;

import me.zeroeightsix.kami.command.Command;
import me.zeroeightsix.kami.command.syntax.ChunkBuilder;

import static me.zeroeightsix.kami.util.MessageSendHelper.sendChatMessage;

/**
 * Created by 086 on 10/10/2018.
 */
public class PrefixCommand extends Command {

    public PrefixCommand() {
        super("prefix", new ChunkBuilder().append("character").build());
        setDescription("Changes the prefix to your new key");
    }

    @Override
    public void call(String[] args) {
        if (args.length <= 0) {
            sendChatMessage("Please specify a new prefix!");
            return;
        }

        if (args[0] != null) {
            Command.commandPrefix.setValue(args[0]);
            sendChatMessage("Prefix set to &b" + Command.commandPrefix.getValue());
        } else if (args[0].equals("\\")) {
            sendChatMessage("Error: \"\\\" is not a supported prefix");
        } else {
            sendChatMessage("Please specify a new prefix!");
        }
    }

}

File: src/main/java/me/zeroeightsix/kami/command/commands/PrefixCommand.kt
package me.zeroeightsix.kami.command.commands

import me.zeroeightsix.kami.command.ClientCommand
import me.zeroeightsix.kami.module.modules.client.CommandConfig
import me.zeroeightsix.kami.util.text.MessageSendHelper
import me.zeroeightsix.kami.util.text.formatValue

object PrefixCommand : ClientCommand(
    name = "prefix",
    description = "Change command prefix"
) {
    init {
        literal("reset") {
            execute("Reset the prefix to ;") {
                CommandConfig.prefix.value = ";"
                MessageSendHelper.sendChatMessage("Reset prefix to [&7;&f]!")
            }
        }

        string("new prefix") { prefixArg ->
            execute("Set a new prefix") {
                if (prefixArg.value.isEmpty() || prefixArg.value == "\\") {
                    CommandConfig.prefix.value = ";"
                    MessageSendHelper.sendChatMessage("Reset prefix to [&7;&f]!")
                    return@execute
                }

                CommandConfig.prefix.value = prefixArg.value
                MessageSendHelper.sendChatMessage("Set prefix to ${formatValue(prefixArg.value)}!")
            }
        }
    }
}